arbos/programs/programs.go

```diff
@@ -308,30 +308,44 @@ func getWasm(statedb vm.StateDB, program common.Address) ([]byte, error) {
 }
 
 func (p Programs) getProgram(contract *vm.Contract) (Program, error) {
-
 	return p.deserializeProgram(contract.CodeHash)
 }
 
 func (p Programs) deserializeProgram(codeHash common.Hash) (Program, error) {
 	data, err := p.programs.Get(codeHash)
+	if err != nil {
+		return Program{}, err
+	}
+	compiledHash, err := p.moduleHashes.Get(codeHash)
```

Comment:
Why not store this inline in data?

Reply:
We could, though `programs` is `storage.Storage` so we'd need to manually read two words or create a new type